jupyter_client/kernelspec.py

```diff
@@ -37,8 +38,25 @@ def from_resource_dir(cls, resource_dir):
         kernel_file = pjoin(resource_dir, 'kernel.json')
         with io.open(kernel_file, 'r', encoding='utf-8') as f:
             kernel_dict = json.load(f)
+        kernel_dict = cls.update_newest_kernel(kernel_file, kernel_dict)
         return cls(resource_dir=resource_dir, **kernel_dict)
 
+    @classmethod
+    def update_newest_kernel(cls, kernel_file, kernel_dict):
+        """Update to the newest kernel.json when kernel"s language is python
+        """
+        if 'language' in kernel_dict and \
+           kernel_dict['language'] == 'python' and \
+           kernel_dict['argv'][0] != sys.executable:
+            kernel_dict['argv'][0] = sys.executable
```

Comment:
This is still not the right thing to do, because it is perfectly valid for there to be many Python kernels pointing to many different Pythons on your system.